yw_website/yw_website/apps/website/utils.py

```diff
@@ -43,3 +45,37 @@ def search_and_create_query_set(q):
         return Workflow.objects.filter(query).exclude(version__isnull=True)
     else:
         return Workflow.objects.none()
+
+def get_block_inputs(run_id):
+    '''
+    This function grabs inputs into the program blocks of a run.
+    KIDDING NO IT DOESN'T
+    This function sets up the block structure so that from a parent program
+    block you may drill down into a child of that program block
+    '''
+    program_blocks = ProgramBlock.objects.filter(run=run_id)
+    x = []
+    parent = ProgramBlock()
```

Comment:
you can find the parent before this for loop by filtering program_blocks again on `in_program_block=None`